libcrux-ml-kem/src/ind_cca.rs

```diff
@@ -268,12 +352,13 @@ pub(crate) fn decapsulate<
         Scheme::kdf::<K, CIPHERTEXT_SIZE, Hasher>(&implicit_rejection_shared_secret, ciphertext);
     let shared_secret = Scheme::kdf::<K, CIPHERTEXT_SIZE, Hasher>(shared_secret, ciphertext);
 
-    compare_ciphertexts_select_shared_secret_in_constant_time(
-        ciphertext.as_ref(),
-        &expected_ciphertext,
-        &shared_secret,
-        &implicit_rejection_shared_secret,
-    )
+    let shared_secret = compare_ciphertexts_select_shared_secret_in_constant_time(
```

Comment:
In this case, it is not needed now. But when verifying functions in F*, we often need to have a name for the return value so that we can assert things about it. So we often systematically name the return value in functions, and then after the full proof is done, sometimes, like here, the assertions on the output value can be deleted.